UserPalletのSwaggerタグをつける

keykimu opened the pull request:
#26 の実装

keykimu committed:
add: Dtoのドキュメント文の追加

keykimu committed:
add: userpalletのentityを追加

keykimu committed:
fix: IsOptional()の削除

keykimu committed:
add: create-user-pallet.dtoの追加

keykimu committed:
add: update-user-palletの追加

keykimu committed:
update:DTOをControllerに適用

keykimu committed:
fix: controllerとdtoにタグ付け

RimlTempest commented on src/common/constants.ts:
```diff
@@ -28,4 +28,8 @@ export class Constants {
   public static readonly VERIFY_STATUS = 'ステータスコード';
   public static readonly VERIFY_MESSAGE = 'ステータスメッセージ';
+
+  //UserPallet DTO
```
//とコメントの間スペース開ける

RimlTempest commented on src/user-pallet/dto/create-user-pallet.dto.ts:
```diff
@@ -1 +1,19 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsString } from 'class-validator';
+import { Exclude } from 'class-transformer';
+export class CreateUserPalletDto {
```
レスポンスで使用するDTOなのかリクエストで使用するDTOなのかわかりにくいのでCreateUserPalletRequestみたいな感じで何に使うかを明確化する

keykimu replied:
確か自動生成されて名前変更してなかったからかも？

RimlTempest replied:
自動生成はあくまでテンプレート吐き出しなので内容は適した形に書き換える必要がある

RimlTempest commented on src/user-pallet/dto/create-user-pallet.dto.ts:
```diff
@@ -1 +1,19 @@
+import { IsString } from 'class-validator';
+import { Exclude } from 'class-transformer';
+export class CreateUserPalletDto {
+  @ApiProperty({ description: 'userId' })
```
descriptionはconstantsに記載するか記載されてる内容を使用する

RimlTempest commented on src/user-pallet/dto/update-user-pallet.dto.ts:
```diff
@@ -1,4 +1,12 @@
+import { IsString } from 'class-validator';
 
-export class UpdateUserPalletDto extends PartialType(CreateUserPalletDto) {}
+export class UpdateUserPalletDto {
```
上記と同じく

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,26 +20,44 @@ export class UserPalletController {
-  create(@Body() data: Prisma.UserPalletCreateInput): Promise<UserPallet> {
-    return this.userPalletService.create(data);
+  // Swagger定義
+  @ApiOperation({ summary: 'ユーザパレット生成' })
```
正常に値が帰ってきた場合のSwaggerドキュメントも追加する

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,26 +20,44 @@ export class UserPalletController {
+  // Swagger定義
+  @ApiOperation({ summary: 'ユーザパレット生成' })
+  // フックメソッド
+  create(@Body() createUserPalletDto: CreateUserPalletDto) {
```
返り値の型宣言も行う

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,26 +20,44 @@ export class UserPalletController {
+  // Swagger定義
+  @ApiOperation({ summary: '全ユーザパレット検索' })
+  // フックメソッド
   findAll(): Promise<UserPallet[]> {
```
返り値の型をDTOで定義したものにする

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,26 +20,44 @@ export class UserPalletController {
+  // Swagger定義
+  @ApiOperation({ summary: 'IDから単一ユーザパレット検索' })
+  // フックメソッド
   findOne(@Param('id') id: string): Promise<UserPallet> {
```
上記と同じく

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,26 +20,44 @@ export class UserPalletController {
+  update(
+    @Param('id') id: string,
+    @Body() updateUserPalletDto: UpdateUserPalletDto,
+  ) {
```
何も値を返却していないのはおかしいのでservice側で定義しているメソッド側でちゃんと返却するようにしこちらで呼ぶ

keykimu committed:
fix: //とコメントの間をあける

keykimu committed:
fix: クラス名の変更とdescription下の修正

keykimu committed:
fix: userpallet.dtoのクラス名変更に伴う修正

keykimu committed:
fix: update-user-pallet.dtoのクラス名を変更

keykimu committed:
fix: update-user-pallet.dtoのクラス名変更に伴う修正

keykimu committed:
fix: userIdはstring型のため修正

keykimu committed:
add: userpalletの全取得のためのdto

keykimu committed:
add: userpalletの単一取得のためのdto

keykimu committed:
fix: userにswaggerタグをつける（#33）を参考に修正

keykimu committed:
fix: レビューの修正

・正常に値が帰ってきた場合のSwaggerドキュメントも追加
・userにswaggerタグをつける(#33)を参考に修正

keykimu committed:
fix: deleteタグを消していたので追記と関数名の修正

RimlTempest commented on src/user-pallet/dto/create-user-pallet.dto.ts:
```diff
@@ -1 +1,21 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty, IsString } from 'class-validator';
+import { Constants } from 'src/common/constants';
+import { UserPallet } from '.prisma/client';
```
prisma/clientを直接使用しないでUserPalletのEntityを定義してそれを扱う

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,27 +28,57 @@ export class UserPalletController {
-    return this.userPalletService.remove(+id);
+  // Swagger定義
+  @ApiOperation({ summary: 'ユーザパレット削除' })
+  // フックメソッド
```
@ApiResponseも定義する

RimlTempest commented on src/user-pallet/user-pallet.controller.ts:
```diff
@@ -18,27 +28,57 @@ export class UserPalletController {
+    return;
   }
 
+  @UseGuards(JwtAuthGuard)
```
ユーザパレット情報は秘匿情報ではないのでログインガードする必要ないと思う

keykimu committed:
fix: prismaを直接使用しないよう修正

keykimu committed:
add: userpalletを削除するdtoが抜けていたので追加

keykimu committed:
fix: 項目の追加とレスポンスを追加

keykimu committed:
fix: 項目の追加とレスポンスを追加

keykimu committed:
fix: userpallet.servisの修正

・prismaを各dto（CreateUserPalletRequest・Response,UpdateUserPalletRequest・Response）で置き換え
・追加したuserpallet削除用のdtoを適用

keykimu committed:
fix: userpallet.controllerの修正

・不要なUseGuardsを削除
・@ApiResponseの記述
・レスポンスで置き換え

RimlTempest commented on src/user-pallet/user-pallet.service.ts:
```diff
@@ -1,16 +1,36 @@
-import { Injectable } from '@nestjs/common';
-import { UserPallet, Prisma } from '@prisma/client';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { UserPallet } from '@prisma/client';
```
Entityを使う

keykimu committed:
fix: userがUserと紐づいてなかったので修正

keykimu committed:
fix: userpalletのentityを指定するよう修正

keykimu committed:
fix: userIdの型の変更とuserIdで紐づけるよう修正

RimlTempest reviewed (approved):
![nichijou2](https://user-images.githubusercontent.com/38300385/138229657-357f78ee-9e48-4276-a452-377214460513.gif)
